vyper/venom/basicblock.py

```diff
@@ -237,6 +241,34 @@ def __repr__(self):
         return json.dumps(self.value)  # escape it
 
 
+@dataclass(frozen=True)
+class MemoryLocation:
+    """Represents a memory location that can be analyzed for aliasing"""
+
+    offset: int = 0
+    size: int = 0
+    is_volatile: bool = False
+
+    def completely_overlaps(self, other: "MemoryLocation") -> bool:
+        if self == FULL_MEMORY_ACCESS:
+            return True
+        if other == FULL_MEMORY_ACCESS:
+            return self == FULL_MEMORY_ACCESS
+        if self == EMPTY_MEMORY_ACCESS or other == EMPTY_MEMORY_ACCESS:
+            return False
+        if self.size <= 0 or other.size <= 0:
```

Comment:
size < 0 means full memory, right? feels a bit weird to handle it separately from `FULL_MEMORY_ACCESS`. also, what do you think about having a proper sentinel instead of `-1`? like `MemoryAccess.UNBOUNDED_ACCESS = object()`

Reply:
```suggestion
        assert self.size > 0 and other.size > 0  # guaranteed by previous checks
```